avm/res/insights/private-link-scope/main.bicep

```diff
@@ -0,0 +1,287 @@
+metadata name = 'Azure Monitor Private Link Scopes'
+metadata description = 'This module deploys an Azure Monitor Private Link Scope.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the private link scope.')
+@minLength(1)
+param name string
+
+@description('''Optional. Specifies the access mode of ingestion or queries through associated private endpoints in scope. For security reasons, it is recommended to use PrivateOnly whenever possible to avoid data exfiltration.
```

Comment:
I'd suggest to move the detailed description and logic to udt.

Reply:
That actually may be a bit tricky as inside the UDT there are 2 parameter that accept this parameter. Should the description be duplicated for either?

Reply:
Agree with @AlexanderSehr I would prefer to avoid duplication

Reply:
To avoid duplication we should use udt to create a nested type.

Reply:
The description has been updated to reflect a similar approach to the initial suggestion. User experience:
![image](https://github.com/Azure/bicep-registry-modules/assets/28486158/a3b80c7e-2796-421d-8eda-143fec888351)

Reply:
The additional description in the type looks good. 
What about the one related to this comment? Is that intended to stay as is?

Reply:
I would like to suggest to keep things as is for now please, at least for migration. From a consumer perspective the info is there and supports the deployment.

Reply:
I tend to agree @eriqua with @ahmadabdalla, at least for the migration. Splitting it in 2 UDTs with case handling is something to carefully design - not to mention that UDTs are optional to begin with. However, it is something we could track in an issue for the owner to discuss & implement later.

Reply:
@ahmadabdalla is the owner so I'll leave the future decision to him :smile: Mine was only a suggestion. Resolving and approving